Ondersteuning voor referentie-implementatie

EduardWitteveen opened the pull request:
@haarlem-dvh Kun je eens controleren of deze uitwisselbaar is met jullie huidige versie?
gebruikte config ga ik seperaat via email naar je versturen

@jacodg Lees je mee? Ik denk dat dit nodig is om de nieuwe versie van de standaarden straks ook te kunnen ondersteunen

Commit:
WIP: Added the new authentication and seperate endpoints

Commit:
WIP: Added the import-script for (meta-)data like example/openzaak-export-catalogus-zaaktypes.zip

Commit:
WIP: Trying to get Open Zaakbrug running with the referentie-implementatie

Commit:
WIP: Nu op een andere manier proberen de bearer op te halen, zonder resultaat

haarlem-dvh committed:
[WIP] Authorization Bearer Token werkt, credentials zijn wle nog hardcoded.

Commit:
WIP: Added authorization session

Commit:
WIP: Make it also working with OpenZaak again

Commit:
WIP: Also put the request-headers in the debug-logging

haarlem-dvh commented:
Hoi Eduard,

Ja, ik zal hier naar kijken. Maar weet niet zeker of dat dit jaar nog gaat lukken.
Ik probeer het!

Groeten,
David

Van: Eduard Witteveen ***@***.***>
Verzonden: zondag 17 december 2023 21:18
Aan: Sudwest-Fryslan/OpenZaakBrug ***@***.***>
CC: David van Hussel ***@***.***>; Mention ***@***.***>
Onderwerp: [Sudwest-Fryslan/OpenZaakBrug] Ondersteuning voor referentie-implementatie (PR #360)


@haarlem-dvh<https://github.com/haarlem-dvh> Kun je eens controleren of deze uitwisselbaar is met jullie huidige versie?
gebruikte config ga ik seperaat via email naar je versturen

@jacodg<https://github.com/jacodg> Lees je mee? Ik denk dat dit nodig is om de nieuwe versie van de standaarden straks ook te kunnen ondersteunen

________________________________
You can view, comment on, or merge this pull request online at:

  https://github.com/Sudwest-Fryslan/OpenZaakBrug/pull/360

Commit Summary

  *   1e15cda<https://github.com/Sudwest-Fryslan/OpenZaakBrug/pull/360/commits/1e15cdac295fdc25c0f43734250c390e5b9420c6> WIP: Added the new authentication and seperate endpoints
  *   a55222e<https://github.com/Sudwest-Fryslan/OpenZaakBrug/pull/360/commits/a55222e245099ab3a312e1dc33e08ba147b07f61> WIP: Added the import-script for (meta-)data like example/openzaak-export-catalogus-zaaktypes.zip
  *   711b069<https://github.com/Sudwest-Fryslan/OpenZaakBrug/pull/360/commits/711b0690695373d814c86d361821a8ed522dc477> WIP: Trying to get Open Zaakbrug running with the referentie-implementatie
  *   1d38b96<https://github.com/Sudwest-Fryslan/OpenZaakBrug/pull/360/commits/1d38b96d974a0dd162b61c56da2b72eb10b32793> WIP: Nu op een andere manier proberen de bearer op te halen, zonder resultaat
  *   a091136<https://github.com/Sudwest-Fryslan/OpenZaakBrug/pull/360/commits/a0911364151429fd0bbc0e0590bbbb5717943c69> [WIP] Authorization Bearer Token werkt, credentials zijn wle nog hardcoded.
  *   66599f0<https://github.com/Sudwest-Fryslan/OpenZaakBrug/pull/360/commits/66599f04ea17fd6fabaa5b9ef363ed262c09a0d8> WIP: Added authorization session
  *   439d1cf<https://github.com/Sudwest-Fryslan/OpenZaakBrug/pull/360/commits/439d1cf4a4c89a496c2e669614ed07eb0bd1ffca> WIP: Make it also working with OpenZaak again
  *   7b9863a<https://github.com/Sudwest-Fryslan/OpenZaakBrug/pull/360/commits/7b9863a34c3ab8394d50bb58f94ffa03b95b0b53> WIP: Also put the request-headers in the debug-logging

File Changes

(33 files<https://github.com/Sudwest-Fryslan/OpenZaakBrug/pull/360/files>)

  *   M .gitignore<https://github.com/Sudwest-Fryslan/OpenZaakBrug/pull/360/files#diff-bc37d034bad564583790a46f19d807abfe519c5671395fd494d8cce506c42947> (3)
  *   A examples/import/Config.py<https://github.com/Sudwest-Fryslan/OpenZaakBrug/pull/360/files#diff-de85a5d25aad9662812c61e70513a8f24e5f59b8eb5326bdd2fb9691568ab15f> (32)
  *   A examples/import/README.md<https://github.com/Sudwest-Fryslan/OpenZaakBrug/pull/360/files#diff-07bfaf27a170e4ea09b87f0b9941a2795fac896699e51516bb701957eacce273> (15)
  *   A examples/import/ZgwImportData.py<https://github.com/Sudwest-Fryslan/OpenZaakBrug/pull/360/files#diff-ede29c0872e48634afdd875d18f3739e5691a0ba7e6307dac1d9f2dcdc89a70c> (354)
  *   A examples/import/ZgwImportData.pyproj<https://github.com/Sudwest-Fryslan/OpenZaakBrug/pull/360/files#diff-0d98f6fb75f178d8772897048f9f153305abdb47fad2651d3e25de0f6648ebe0> (57)
  *   A examples/import/ZgwImportData.sln<https://github.com/Sudwest-Fryslan/OpenZaakBrug/pull/360/files#diff-5ca32675f36dceabd52c58750e1185169fe5d8ffaf77c7da0da6645c213a39d8> (23)
  *   A examples/import/logo.png<https://github.com/Sudwest-Fryslan/OpenZaakBrug/pull/360/files#diff-fefb3633a11dbdf340c00020780bd3b4a6038d4298d75fa07d9e2cadd0faa344> (0)
  *   A examples/import/requirements.txt<https://github.com/Sudwest-Fryslan/OpenZaakBrug/pull/360/files#diff-75a976d3ba7cb3c906455320ae3ed21df2f5df1b463970670bb52d8ab9909fd5> (10)
  *   M examples/soap/Open-Zaakbrug-soapui-project.xml<https://github.com/Sudwest-Fryslan/OpenZaakBrug/pull/360/files#diff-99c67338b83b8ed0282b95c67d85fb80ef0ac047c09306d463d953114a9e1a10> (16)
  *   M src/main/java/nl/haarlem/translations/zdstozgw/controller/JWTController.java<https://github.com/Sudwest-Fryslan/OpenZaakBrug/pull/360/files#diff-b68049e188f1ffb6794d36a46ce0a4063d7d6a20f01b1f027005f70443d7693a> (3)
  *   M src/main/java/nl/haarlem/translations/zdstozgw/converter/impl/replicate/ActualiseerZaakStatusReplicator.java<https://github.com/Sudwest-Fryslan/OpenZaakBrug/pull/360/files#diff-a074c515536957c4bef6f74099a0d601e50f69bf42af8cff9f3287fd68ad2031> (6)
  *   M src/main/java/nl/haarlem/translations/zdstozgw/converter/impl/replicate/GeefLijstZaakdocumentenReplicator.java<https://github.com/Sudwest-Fryslan/OpenZaakBrug/pull/360/files#diff-8b8ee8efb1843d22c1506343e8a4cfaf43f3893eef82e5bbd38fd9c67f27e3a9> (5)
  *   M src/main/java/nl/haarlem/translations/zdstozgw/converter/impl/replicate/GeefZaakDetailsReplicator.java<https://github.com/Sudwest-Fryslan/OpenZaakBrug/pull/360/files#diff-d790ba60425c8b58de7ee413591619239e89d81dc19d4de46ec0aa3e6c282d14> (5)
  *   M src/main/java/nl/haarlem/translations/zdstozgw/converter/impl/replicate/GeefZaakdocumentLezenReplicator.java<https://github.com/Sudwest-Fryslan/OpenZaakBrug/pull/360/files#diff-57d6c6887fa50cbbf9d791aa416cf74776ce294782ef5ff5c99614e92a4a8627> (5)
  *   M src/main/java/nl/haarlem/translations/zdstozgw/converter/impl/replicate/Replicator.java<https://github.com/Sudwest-Fryslan/OpenZaakBrug/pull/360/files#diff-c46c55e6cf990c62ed686b170ae8e0e42ef932460c7e8a8383c47226bbb95efb> (45)
  *   M src/main/java/nl/haarlem/translations/zdstozgw/converter/impl/replicate/UpdateZaakReplicator.java<https://github.com/Sudwest-Fryslan/OpenZaakBrug/pull/360/files#diff-0df79e74642966de62ed6c9d68db334a413d33e50bd1a27fe2934eb924d613fb> (5)
  *   M src/main/java/nl/haarlem/translations/zdstozgw/converter/impl/replicate/VoegZaakdocumentToeReplicator.java<https://github.com/Sudwest-Fryslan/OpenZaakBrug/pull/360/files#diff-cead3f22e5d03f5826616ec1b930a3272e929c45af3802bdbd147285b0d859ed> (5)
  *   M src/main/java/nl/haarlem/translations/zdstozgw/converter/impl/translate/ActualiseerZaakStatusTranslator.java<https://github.com/Sudwest-Fryslan/OpenZaakBrug/pull/360/files#diff-5cc852ca19ea99571edf2faec5c1b9b1f5012b2f5813b193aa013f7f018241b7> (5)
  *   M src/main/java/nl/haarlem/translations/zdstozgw/converter/impl/translate/CancelCheckoutTranslator.java<https://github.com/Sudwest-Fryslan/OpenZaakBrug/pull/360/files#diff-502a798a7a765b21ab0e0f67c2c54972307269e4b52bc1550fe95446f73a4c47> (5)
  *   M src/main/java/nl/haarlem/translations/zdstozgw/converter/impl/translate/CreeerZaakTranslator.java<https://github.com/Sudwest-Fryslan/OpenZaakBrug/pull/360/files#diff-bbae91beb91d67b2eec1d855d2a17bc3afca27678a02b04265ba8250357a5b7c> (4)
  *   M src/main/java/nl/haarlem/translations/zdstozgw/converter/impl/translate/GeefLijstZaakdocumentenTranslator.java<https://github.com/Sudwest-Fryslan/OpenZaakBrug/pull/360/files#diff-ddb2b141c535d22e8d22b0d2706905ffbdda411123e1cd94b11384ffa3a45e76> (6)
  *   M src/main/java/nl/haarlem/translations/zdstozgw/converter/impl/translate/GeefZaakDetailsTranslator.java<https://github.com/Sudwest-Fryslan/OpenZaakBrug/pull/360/files#diff-f82317f095a23d3f0e28e8cd188bf9146293df2f79a6909531f62554aeec28b0> (7)
  *   M src/main/java/nl/haarlem/translations/zdstozgw/converter/impl/translate/GeefZaakdocumentBewerkenTranslator.java<https://github.com/Sudwest-Fryslan/OpenZaakBrug/pull/360/files#diff-b722319be21081718279a47873e1f06709476bb6068a32813d12a552843f498f> (8)
  *   M src/main/java/nl/haarlem/translations/zdstozgw/converter/impl/translate/GeefZaakdocumentLezenTranslator.java<https://github.com/Sudwest-Fryslan/OpenZaakBrug/pull/360/files#diff-694b32483eb0c4974206e18b5747eac21b971f54688adba8b08ce3ca3e75b852> (5)
  *   M src/main/java/nl/haarlem/translations/zdstozgw/converter/impl/translate/UpdateZaakTranslator.java<https://github.com/Sudwest-Fryslan/OpenZaakBrug/pull/360/files#diff-a2c0742b4be1c0714b365031650b7c09196229f0192f935db6d30e2ecf37817e> (5)
  *   M src/main/java/nl/haarlem/translations/zdstozgw/converter/impl/translate/UpdateZaakdocumentTranslator.java<https://github.com/Sudwest-Fryslan/OpenZaakBrug/pull/360/files#diff-da36d7bb940892ca00a71c2f1e61246609569d697911398f6f6b73695d3b964f> (5)
  *   M src/main/java/nl/haarlem/translations/zdstozgw/converter/impl/translate/VoegZaakdocumentToeTranslator.java<https://github.com/Sudwest-Fryslan/OpenZaakBrug/pull/360/files#diff-bb974a58a81d90e04efc4788f121fdb4394b69b562077d2133f075c8c3991be7> (6)
  *   M src/main/java/nl/haarlem/translations/zdstozgw/translation/zds/services/ZaakService.java<https://github.com/Sudwest-Fryslan/OpenZaakBrug/pull/360/files#diff-3b1b779a9e56cf7caba088d9a4bdac34740a7ccf96c7b197628f1df7a140b02d> (253)
  *   M src/main/java/nl/haarlem/translations/zdstozgw/translation/zgw/client/JWTService.java<https://github.com/Sudwest-Fryslan/OpenZaakBrug/pull/360/files#diff-3cb61519210464125925707f1df10d024daa11831e999c9be2b03409c278c6d4> (22)
  *   M src/main/java/nl/haarlem/translations/zdstozgw/translation/zgw/client/RestTemplateService.java<https://github.com/Sudwest-Fryslan/OpenZaakBrug/pull/360/files#diff-44fdfaed49c3916a5ba69e175b25b47c25918c4476a7dc887e3ad82adf85d094> (8)
  *   M src/main/java/nl/haarlem/translations/zdstozgw/translation/zgw/client/ZGWClient.java<https://github.com/Sudwest-Fryslan/OpenZaakBrug/pull/360/files#diff-6223ce9063c7123a49487aacd7afe6b88b68c756209e11d52a6fe6bc10344be2> (513)
  *   A src/main/java/nl/haarlem/translations/zdstozgw/translation/zgw/model/ZgwAuthorization.java<https://github.com/Sudwest-Fryslan/OpenZaakBrug/pull/360/files#diff-ecb836758ec1bdab13f4ab77f44502aba79e034822465f516cdd06d45b307f3e> (10)
  *   A src/main/java/nl/haarlem/translations/zdstozgw/translation/zgw/model/ZgwCatalogus.java<https://github.com/Sudwest-Fryslan/OpenZaakBrug/pull/360/files#diff-6d0df30c136d4c34f953d571867dbb123e8850226132219d54d19c5cb35b0f64> (39)

Patch Links:

  *   https://github.com/Sudwest-Fryslan/OpenZaakBrug/pull/360.patch
  *   https://github.com/Sudwest-Fryslan/OpenZaakBrug/pull/360.diff

—
Reply to this email directly, view it on GitHub<https://github.com/Sudwest-Fryslan/OpenZaakBrug/pull/360>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AHWTHXSZOVGK2BLRGHT3YILYJ5HOTAVCNFSM6AAAAABAYSZ3FGVHI2DSMVQWIX3LMV43ASLTON2WKOZSGA2DKMZWGI4DCNQ>.
You are receiving this because you were mentioned.Message ID: ***@***.******@***.***>>

Commit:
Merge remote-tracking branch 'origin/master' into vng-referentie-impl

Commit:
WIP: Support for different JwtAuthorization for each endpoint

Commit:
Open Zaakbrug now supports:
- ZgwReferentie implementatie
- OpenZaak

Commit:
sample config for openzaak

EduardWitteveen committed:
Merge branch 'master' into vng-referentie-impl

Commit:
WIP: get the tests working again

Commit:
Merge branch 'vng-referentie-impl' of https://github.com/Sudwest-Fryslan/OpenZaakBrug into vng-referentie-impl

Commit:
Testen moeten nu ook weer werken!

Commit:
a little bit refactoring

EduardWitteveen commented:
@jacodg Eigenlijk moet de ladybug ook nog de jwt-authenticatie meenemen denk ik, moeten we dan hier wat doen? 
https://github.com/Sudwest-Fryslan/OpenZaakBrug/pull/360/files#diff-77bd0a5da83b73cb71c60d1664287e9812229ab22e9763714901f2af1b07ea17R100

@haarlem-dvh Heb je alweer wat tijd :-)

Commit:
spelling mistake

Commit:
- Post of EnkelvoudigInformatieObject contained a lock, this is not the wanted behaviour
- Config.json can now contain a default organisation, which will be used then not provided by the request

Commit:
- solved some null-pointer exceptions

EduardWitteveen committed:
Merge branch 'master' into vng-referentie-impl

haarlem-dvh commented on src/main/java/nl/haarlem/translations/zdstozgw/translation/zgw/client/ZgwAuthorization.java:
```diff
@@ -0,0 +1,139 @@
+		// do we already have an endpoint for this jwt-token?
+		for(ZgwJwtTokenEntry entry : this.authorizations.values()) {			
+			if (entry.jwtUrl == null) {
+				if(jwtUrl != null) continue;
```
Dit kan in één if statement, toch?

haarlem-dvh commented on src/main/java/nl/haarlem/translations/zdstozgw/translation/zgw/client/ZgwAuthorization.java:
```diff
@@ -0,0 +1,139 @@
+				continue;
+			}
+			// bestond al, geven we terug
+			log.debug("For url: " + url + " using: '" + entry.getAuthorization() +  "' from previous found entry");
```
Dit word toch juist geloged als hij nog niet bestaat? Je gaat hem daarna ook toevoegen.

haarlem-dvh commented on src/main/java/nl/haarlem/translations/zdstozgw/translation/zgw/model/ZgwEnkelvoudigInformatieObjectPost.java:
```diff
@@ -0,0 +1,113 @@
+	@SerializedName("informatieobjecttype")
+	@Expose
+	public String informatieobjecttype;			
+//	@SerializedName("versie")
```
Deze regels mogen weg.

haarlem-dvh commented on src/main/java/nl/haarlem/translations/zdstozgw/config/ModelMapperConfig.java:
```diff
@@ -122,10 +123,14 @@ public ModelMapper modelMapper() {
-		addZdsZaakDocumentInhoudToZgwEnkelvoudigInformatieObjectTypeMapping(modelMapper);
+		
+		addZdsZaakDocumentInhoudToZgwEnkelvoudigInformatieObjectPostTypeMapping(modelMapper);
+		// is de volgende nog nodig?
```
Blijkbaar niet, als alle testen goed gaan ;-)

haarlem-dvh commented on src/main/java/nl/haarlem/translations/zdstozgw/config/ModelMapperConfig.java:
```diff
@@ -229,19 +234,43 @@ public void addZgwEnkelvoudigInformatieObjectToZdsZaakDocumentInhoudTypeMapping(
+	
+	
+
+//	public void addZdsZaakDocumentInhoudToZgwEnkelvoudigInformatieObjectTypeMapping(ModelMapper modelMapper) {
```
Dan mag hij hier ook helemaal weg.

haarlem-dvh reviewed (changes requested):
Hoi Eduard, 
Helaas heb ik nog geen tijd gehad om deze branch lokaal tegen onze omgeving te testen.
Maar heb wel naar de code aanpassingen gekeken. Op paar kleine puntjes na volgens mij helemaal prima. 

Groeten,
David

Commit:
- UpdateZaakDocument werkend gemaakt

Commit:
- updated the testcases

Commit:
- updated the SoapUI test project

DelanoWAF committed:
build: update spring-boot

DelanoWAF committed:
build: add missing dependency version

Commit:
- updated the soapui-tests

DelanoWAF committed:
build: update spring boot to 2.7.18

Commit:
Master erbij in gemerged

Commit:
- make all soapui test working
- fixed null-pointer exception
- better errormessages

Commit:
Merge branch '328-Make-CLO-Working' of https://github.com/Sudwest-Fryslan/OpenZaakBrug into vng-referentie-impl

Commit:
Merge branch 'spring-boot-update' of https://github.com/Sudwest-Fryslan/OpenZaakBrug into vng-referentie-impl

Commit:
- ignore (more) white-spaces in xml

Commit:
- we want the same output as in the master branche

Commit:
- improved the import-zgw-data script

Commit:
improved zgw-data import

Commit:
better errror resolving:
- upon strange build-errors, the first will point to missing lombok
- better error message when config.json is missing

Commit:
Merge branch 'vng-referentie-impl' of https://github.com/Sudwest-Fryslan/OpenZaakBrug into vng-referentie-impl

Commit:
- updated zaaktypes en tests

Commit:
- added profportaal berichten

Commit:
- created 4 test cases for performance measurement

Commit:
Merge remote-tracking branch 'origin/master' into vng-referentie-impl

# Conflicts:
#	pom.xml

Commit:
Update dev-hulptekst, testconfig en verduidelijk verouderde ID-parameters

- RequestHandler.java: concrete downloadlink/commando toegevoegd bij de lombok-IDE-foutmelding
- ModelMapperConfig.java: comment toegevoegd waarom StUF-tijdstempels hardcoded Europe/Amsterdam gebruiken
- data-h2.sql / data-postgresql.sql: parameter_description van ZaakIdentificatieHuidige en
  DocumentIdentificatieHuidige markeert nu dat deze waarden niet meer worden gebruikt sinds de
  introductie van sequences, om verwarring bij toekomstig aanpassen te voorkomen
- soapui-project.xml: bijgewerkte test-ID's en lokaal serveradres
